libraries/AP_AccelCal/AP_AccelCal.cpp

```diff
@@ -0,0 +1,361 @@
+}
+
+// get the calibrator associated with the ID, ID is given to clients in the order of there
+// registration
+AccelCalibrator* AP_AccelCal::get_calibrator(uint8_t index) {
+    AccelCalibrator* ret;
+    for(uint8_t i=0; i<_num_clients; i++) {
+        for(uint8_t j=0 ; (ret = _clients[i]->_acal_get_calibrator(j)) ; j++) {
+            if (index == 0) {
+                return ret;
+            }
+            index--;
+        }
+    }
+    return NULL;
+}
+
+// update the state of the Accel calibrator server
+void AP_AccelCal::update_status() {
+    AccelCalibrator *cal;
+
+    if (!get_calibrator(0)) {
+        // no calibrators
+        _status = ACCEL_CAL_NOT_STARTED;
+        return;
+    }
+
+    for(uint8_t i=0 ; (cal = get_calibrator(i))  ; i++) {
+        if (cal->get_status() == ACCEL_CAL_FAILED) {
+            _status = ACCEL_CAL_FAILED;         //fail if even one of the calibration has
+            return;
+        }
+    }
+
+    for(uint8_t i=0 ; (cal = get_calibrator(i))  ; i++) {
+        if (cal->get_status() == ACCEL_CAL_COLLECTING_SAMPLE) {
+            _status = ACCEL_CAL_COLLECTING_SAMPLE;          // move to Collecting sample state if all the callibrators have
+            return;
+        }
+    }
+
+    for(uint8_t i=0 ; (cal = get_calibrator(i))  ; i++) {
+        if (cal->get_status() == ACCEL_CAL_WAITING_FOR_ORIENTATION) {
+            _status = ACCEL_CAL_WAITING_FOR_ORIENTATION;    // move to waiting for user ack for orientation confirmation
+            return;
+        }
+    }
+
+    for(uint8_t i=0 ; (cal = get_calibrator(i))  ; i++) {
+        if (cal->get_status() == ACCEL_CAL_NOT_STARTED) {
+            _status = ACCEL_CAL_NOT_STARTED;    // we haven't started if all the calibrators haven't
+            return;
+        }
+    }
+
+    _status = ACCEL_CAL_SUCCESS;    // we have succeeded calibration if all the calibrators have
+    return;
+}
+
+// check if client's calibrator is active
+bool AP_AccelCal::client_active(uint8_t client_num)
+{
+    return (bool)_clients[client_num]->_acal_get_calibrator(0);
+}
+
+/*
+  watch for COMMAND_ACK messages
+ */
+static void _snoop(const mavlink_message_t* msg)
+{
+    if (msg->msgid == MAVLINK_MSG_ID_COMMAND_ACK) {
+        _start_collect_sample = true;
+    }
+}
+
+void AP_AccelCal::_printf(const char* fmt, ...)
+{
+    if (!_gcs) {
+        return;
+    }
+    char msg[50];
+    va_list ap;
+    va_start(ap, fmt);
+    hal.util->vsnprintf(msg, sizeof(msg), fmt, ap);
+    va_end(ap);
+    if (msg[strlen(msg)-1] == '\n') {
+        // STATUSTEXT messages should not add linefeed
+        msg[strlen(msg)-1] = 0;
+    }
+    AP_HAL::UARTDriver *uart = _gcs->get_uart();
+    /*
+     *     to ensure these messages get to the user we need to wait for the
+     *     port send buffer to have enough room
+     */
+    while (uart->txspace() < MAVLINK_NUM_NON_PAYLOAD_BYTES+MAVLINK_MSG_ID_STATUSTEXT_LEN) {
+        hal.scheduler->delay(1);
+    }
+#if !REPLAY
```

Comment:
"REPLAY" is far too generic an ifdef to use for something like this. It could well be defined by some system library. 
If you really need different behaviour for replay then see the libraries/AP_Vehicle/AP_Vehicle_Type.h macros